Tests/PenguinStructuresTests/TupleTests.swift

```diff
@@ -0,0 +1,167 @@
+    XCTAssertNotEqual(Tuple(1, "2"), Tuple(1, "XXX"))
+    
+    XCTAssertEqual(Tuple(1, 2.3, "4"), Tuple(1, 2.3, "4"))
+    XCTAssertNotEqual(Tuple(1, 2.3, "5"), Tuple(1, 2.3, "4"))
+    XCTAssertNotEqual(Tuple(1, 2.9, "4"), Tuple(1, 2.3, "4"))
+    XCTAssertNotEqual(Tuple(0, 2.3, "4"), Tuple(1, 2.3, "4"))
+    // This effectively tests Hashable too; we know the conformance synthesizer
+    // is considering all the fields in order.
+  }
+
+  func test_Comparable() {
+    XCTAssertLessThanOrEqual(Tuple0(), Tuple0())
+    XCTAssertGreaterThanOrEqual(Tuple0(), Tuple0())
+    
+    XCTAssertLessThan(Tuple(1), Tuple(2))
+    // Consistency with equality
+    XCTAssertLessThanOrEqual(Tuple(2), Tuple(2))
+    XCTAssertGreaterThanOrEqual(Tuple(2), Tuple(2))
+
+    XCTAssertLessThan(Tuple(1, 0.1), Tuple(2, 1.1))
+    XCTAssertLessThan(Tuple(1, 1.1), Tuple(2, 1.1))
+    XCTAssertLessThan(Tuple(2, 0.1), Tuple(2, 1.1))
+    
+    // Consistency with equality
+    XCTAssertLessThanOrEqual(Tuple(2, 1.1), Tuple(2, 1.1))
+    XCTAssertGreaterThanOrEqual(Tuple(2, 1.1), Tuple(2, 1.1))
+
+    XCTAssertLessThan(Tuple(1, 0.1, 0 as UInt), Tuple(2, 1.1, 1 as UInt))
+    XCTAssertLessThan(Tuple(1, 0.1, 1 as UInt), Tuple(2, 1.1, 1 as UInt))
+    XCTAssertLessThan(Tuple(1, 1.1, 0 as UInt), Tuple(2, 1.1, 1 as UInt))
+    XCTAssertLessThan(Tuple(1, 1.1, 1 as UInt), Tuple(2, 1.1, 1 as UInt))
+    XCTAssertLessThan(Tuple(2, 0.1, 0 as UInt), Tuple(2, 1.1, 1 as UInt))
+    XCTAssertLessThan(Tuple(2, 0.1, 1 as UInt), Tuple(2, 1.1, 1 as UInt))
+    XCTAssertLessThan(Tuple(2, 1.1, 0 as UInt), Tuple(2, 1.1, 1 as UInt))
+    
+    // Consistency with equality
+    XCTAssertLessThanOrEqual(Tuple(2, 1.1, 1 as UInt), Tuple(2, 1.1, 1 as UInt))
+    XCTAssertGreaterThanOrEqual(Tuple(2, 1.1, 1 as UInt), Tuple(2, 1.1, 1 as UInt))
+  }
+
+  func test_CustomStringConvertible() {
+    XCTAssertEqual("\(Empty())", "Empty()")
+    XCTAssertEqual("\(Tuple(1))", "Tuple(1)")
+    XCTAssertEqual("\(Tuple(1, 2.5, "foo"))", "Tuple(1, 2.5, \"foo\")")
+  }
+
+  func test_conveniences() {
+    struct X0 {  }; let x0 = X0()
+    struct X1 {  }; let x1 = X1()
+    struct X2 {  }; let x2 = X2()
+    struct X3 {  }; let x3 = X3()
+    struct X4 {  }; let x4 = X4()
+    struct X5 {  }; let x5 = X5()
+    struct X6 {  }; let x6 = X6()
+
+    XCTAssert(type(of: Tuple(x0)) == Tuple1<X0>.self)
+    XCTAssert(type(of: Tuple(x0, x1)) == Tuple2<X0, X1>.self)
+    XCTAssert(type(of: Tuple(x0, x1, x2)) == Tuple3<X0, X1, X2>.self)
+    XCTAssert(type(of: Tuple(x0, x1, x2, x3)) == Tuple4<X0, X1, X2, X3>.self)
+    XCTAssert(
+      type(of: Tuple(x0, x1, x2, x3, x4)) == Tuple5<X0, X1, X2, X3, X4>.self)
+    XCTAssert(
+      type(of: Tuple(x0, x1, x2, x3, x4, x5))
+        == Tuple6<X0, X1, X2, X3, X4, X5>.self)
+    XCTAssert(
+      type(of: Tuple(x0, x1, x2, x3, x4, x5, x6))
+        == Tuple7<X0, X1, X2, X3, X4, X5, X6>.self)
+  }
+  
+  static var allTests = [
+    ("test_mapReduce", test_mapReduce),
+  ]
```

Comment:
```suggestion
  static var allTests = [
    ("test_mapReduce", test_mapReduce),
    ("test_head", test_head),
    ("test_tail", test_tail),
    ("test_count", test_count),
    ("test_DefaultInitializable", test_DefaultInitializable),
    ("test_Equatable", test_Equatable),
    ("test_Comparable", test_Comparable),
    ("test_CustomStringConvertible", test_CustomStringConvertible),
    ("test_conveniences", test_conveniences),
  ]
```

Reply:
Agh, thanks.  I knew I was going to forget that!